src/opendr/engine/target.py

```diff
@@ -64,9 +119,61 @@ def __init__(self, prediction: int, description=None, confidence=None):
                 One-dimensional array / tensor of class probabilities. Defaults to None.
         """
         super().__init__()
-        self.data = prediction
-        self.confidence = confidence
-        self.description = description
+        self._description = None
+
+        if prediction is not None:
+            self.data = prediction
+
+        if description is not None:
+            self.description = description
+
+        if confidence is not None:
+            self.confidence = confidence
```

Comment:
Is this correct or should it be
```suggestion
        if prediction is not None:
            self._data = prediction

        if description is not None:
            self._description = description

        if confidence is not None:
            self._confidence = confidence
```

Reply:
This is intentional in order to perform type checking through the setter.